packages/tznft/src/ipfs.ts

```
@@ -1,36 +1,36 @@
 import { pinFile, pinDirectory } from '@oxheadalpha/tezos-tools';
-import { loadUserConfig } from './config-util';
+import { loadConfig, saveConfig } from './config';
 import * as kleur from 'kleur';
 import * as path from 'path';
 import * as fs from 'fs';
 
-const ipfsKey = 'ipfs-pinata';
-
 export async function setPinataKeys(
   apiKey: string,
   secretKey: string,
   force: boolean
 ) {
   console.log(apiKey, secretKey, force);
-  const config = loadUserConfig();
-  if (config.has(ipfsKey) && !force) {
+  const config = await loadConfig();
+  config.pinataIpfs;
+  if (config.pinataIpfs && !force) {
     console.log(
       kleur.red('Pinata keys already exist. Use --force option to override.')
     );
     return;
   }
-  config.set(ipfsKey, { apiKey, secretKey });
+  config.pinataIpfs = { apiKey, secretKey };
+  saveConfig(config);
   console.log(kleur.green('Pinata keys have been added.'));
 }
 
 type PinataKeys = { apiKey: string; secretKey: string };
 
 export async function pinFileToIpfs(tag: string, filePath: string) {
-  const pinataKeys = loadPinataKeys();
-  if(!pinataKeys) return;
+  const pinataKeys = await loadPinataKeys();
+  if (!pinataKeys) return;
```

Consider print a warning to console that the command cannot be completed.
Ditto all other places in `pinFileToIpfs()` and `pinDirectoryToIpfs()` that silently return w/o actual pinning

---

I did not change your logic, just substituted with a different config. I believe, we should give a warning.

---

Yes, original implementation should be improved

---

Left it as it was, `loadPinataKeys` inside prints a warning into the console.